jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -385,6 +419,206 @@ public Integer getMaxQueueSize() {
+          log.debug("Using HTTP Basic authentication with data from the environment variables.");
+          authInterceptor = getAddBasicAuthInterceptor(collectorAuthUsername, collectorAuthPassword);
+        }
+
+        // at this point, the authInterceptor can still be null
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+
+        if (null != authInterceptor) {
+          log.debug("All requests to the Jaeger Collector will have an Authentication header set.");
+          builder.addInterceptor(authInterceptor);
+        }
+
+        log.debug("Using the HTTP Sender to send spans directly to the Jaeger Collector.");
+        return new HttpSender(collectorEndpoint, builder.build());
+      }
+
+      log.debug("Using the UDP Sender to send spans to the Jaeger Agent.");
+      return new UdpSender(
+              stringOrDefault(this.agentHost, UdpSender.DEFAULT_AGENT_UDP_HOST),
+              numberOrDefault(this.agentPort, UdpSender.DEFAULT_AGENT_UDP_COMPACT_PORT).intValue(),
+              0 /* max packet size */);
+    }
+
+    public String getAgentHost() {
+      return agentHost;
+    }
+
+    public Integer getAgentPort() {
+      return agentPort;
+    }
+
+    public String getCollectorEndpoint() {
+      return collectorEndpoint;
+    }
+
+    public String getCollectorAuthToken() {
+      return collectorAuthToken;
+    }
+
+    public String getCollectorAuthUsername() {
+      return collectorAuthUsername;
+    }
+
+    public String getCollectorAuthPassword() {
+      return collectorAuthPassword;
+    }
+
+    /**
+     * Attempts to create a new {@link SenderConfiguration} based on the environment variables
+     * @return a new sender configuration based on environment variables
+     */
+    public static SenderConfiguration fromEnv() {
+      String agentHost = getProperty(JAEGER_AGENT_HOST);
+      Integer agentPort = getPropertyAsInt(JAEGER_AGENT_PORT);
+
+      String collectorEndpoint = getProperty(JAEGER_COLLECTOR_ENDPOINT);
+      String authToken = getProperty(JAEGER_COLLECTOR_AUTH_TOKEN);
+      String authUsername = getProperty(JAEGER_COLLECTOR_REMOTE_USER);
+      String authPassword = getProperty(JAEGER_COLLECTOR_REMOTE_PASSWORD);
+
+      return new SenderConfiguration(agentHost, agentPort, collectorEndpoint, authToken, authUsername, authPassword);
+    }
+
+    private Interceptor getAddTokenInterceptor(final String authToken) {
+      return getInterceptor("Bearer " + authToken);
+    }
+
+    private Interceptor getAddBasicAuthInterceptor(final String username, final String password) {
+      return getInterceptor(Credentials.basic(username, password));
+    }
+
+    private Interceptor getInterceptor(final String headerValue) {
+      return new Interceptor() {
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+          return chain.proceed(
+                  chain
```

Comment:
This formatting looks weird

Reply:
I think it's formatted according to the checkstyle. The alternative is to create a var to hold the inner part.